help_function/src/util_group_store_test.cpp

```diff
@@ -0,0 +1,305 @@
+          }
+        });
+  });
+  q.wait_and_throw();
+
+  sycl::host_accessor data_accessor(buffer_out, sycl::read_only);
+  const int *ptr = data_accessor.get_multi_ptr<sycl::access::decorated::yes>();
+  sycl::host_accessor data_accessor_sg(sg_sz_buf, sycl::read_only);
+  const uint32_t *ptr_sg =
+      data_accessor_sg.get_multi_ptr<sycl::access::decorated::yes>();
+  return subgroup_helper_validation_function(
+      ptr, ptr_sg, "test_subgroup_striped_standalone");
+}
+
+
+
+template <dpct::group::store_algorithm S> bool test_group_store_standalone() {
+  // Tests dpct::group::load_algorithm::BLOCK_LOAD_DIRECT &
+  // dpct::group::load_algorithm::BLOCK_LOAD_STRIPED as standalone methods
+  // Tests dpct::group::store_algorithm::BLOCK_STORE_DIRECT &
+  // dpct::group::store_algorithm::BLOCK_STORE_STRIPED as standalone methods
+  sycl::queue q(dpct::get_default_queue());
+  int data[512];
+  int num_threads = 128;
+  int items_per_thread = 4;
+  if constexpr(S == dpct::group::store_algorithm::BLOCK_STORE_DIRECT) {
+    for (int i = 0; i < 512; i++) {data[i] = i;}
+  }
+  else{
+    for (int i = 0; i < num_threads; ++i) {
+      for (int j = 0; j < items_per_thread; ++j) {
+        data[i * items_per_thread + j] = j * num_threads + i;  
+      }
+      
+    }
+  }
+  
+  std::cout<<std::endl;
+  int data_out[512];
+  sycl::buffer<int, 1> buffer(data, 512);
+  sycl::buffer<int, 1> buffer_out(data_out, 512);
+  
+  
+  q.submit([&](sycl::handler &h) {
+    sycl::accessor dacc_read(buffer, h, sycl::read_write);
+    sycl::accessor dacc_write(buffer_out, h, sycl::read_write);
+    h.parallel_for(
+        sycl::nd_range<3>(sycl::range<3>(1, 1, 128), sycl::range<3>(1, 1, 128)),
+        [=](sycl::nd_item<3> item) {
+          int thread_data[4];
+          int global_index =
+              item.get_group(2) * item.get_local_range().get(2) +
+              item.get_local_id(2); 
+          #pragma unroll
+          for (int i = 0; i < 4; ++i) {
+            thread_data[i] = dacc_read[global_index * 4 + i];
+          }
+        
+          auto *d_r =
+              dacc_read.get_multi_ptr<sycl::access::decorated::yes>().get();
+          auto *d_w =
+              dacc_write.get_multi_ptr<sycl::access::decorated::yes>().get();
+          // Store thread_data of each work item from blocked arrangement
+          if (S == dpct::group::store_algorithm::BLOCK_STORE_DIRECT) {
+            dpct::group::store_blocked<4, int>(item, d_r, thread_data);
+          } else {
+            dpct::group::store_striped<4, int>(item, d_r, thread_data);
+          }
+          // Write thread_data of each work item at index to the global buffer
+          
+          global_index =
+              item.get_group(2) * item.get_local_range().get(2) +
+              item.get_local_id(2);  // Each thread_data has 4 elements
+          #pragma unroll
+          for (int i = 0; i < 4; ++i) {
+            dacc_write[global_index * 4 + i] = thread_data[i];

```

Comment:
Similar question to above about the purpose of this and how we are verifying the store routines at all.

Reply:
Since the recent changes, we are now doing `group_store(tmp).store(item, d_w, thread_data);` where `d_w` is referring to `buffer_out`. This part seems correct, but my comment about these lines still apply.

It seems like the contents of `buffer_out` after the `group_store` operation are being overwritten by the writes in this loop. Can we just remove this entire loop? I am not sure why it is here unless I misunderstand.

Reply:
Yes modified to prevent the overwrite, could you please take a look . Thanks

Reply:
Thanks, this looks to resolve the issue